walk/walk.go

```diff
@@ -170,24 +178,39 @@ func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnored
 		case ent == nil:
 			continue
 		case ent.IsDir():
-			subdirs = append(subdirs, base)
+			dirs = append(dirs, base)
 		default:
 			regularFiles = append(regularFiles, base)
 		}
 	}
 
+	var subdirs []string
+
+	// Visit subdirectories dividing them into `regularFiles` and `subdirs``
 	shouldUpdate := updateRels.shouldUpdate(rel, updateParent)
-	for _, sub := range subdirs {
+	for _, sub := range dirs {
 		if subRel := path.Join(rel, sub); updateRels.shouldVisit(subRel, shouldUpdate) {
-			visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, filepath.Join(dir, sub), subRel, shouldUpdate)
+			subFiles, mergeFiles := visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, filepath.Join(dir, sub), subRel, shouldUpdate)
```

Comment:
nit: maybe rename mergeFiles to `shouldMerge` or `isInParent` or something that makes it more obviously a boolean?